server/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java

```diff
@@ -187,6 +187,25 @@ public ShardSearchRequest(StreamInput in) throws IOException {
         originalIndices = OriginalIndices.readOriginalIndices(in);
     }
 
+    public ShardSearchRequest(ShardSearchRequest clone) {
+        this.shardId = clone.shardId;
+        this.searchType = clone.searchType;
+        this.numberOfShards = clone.numberOfShards;
+        this.scroll = clone.scroll;
+        this.source = clone.source;
+        this.aliasFilter = clone.aliasFilter;
+        this.indexBoost = clone.indexBoost;
+        this.nowInMillis = clone.nowInMillis;
+        this.requestCache = clone.requestCache;
+        this.clusterAlias = clone.clusterAlias;
+        this.allowPartialSearchResults = clone.allowPartialSearchResults;
+        this.indexRoutings = clone.indexRoutings;
+        this.preference = clone.preference;
+        this.canReturnNullResponseIfMatchNoDocs = clone.canReturnNullResponseIfMatchNoDocs;
+        this.bottomSortValues = clone.bottomSortValues;
+        this.originalIndices = clone.originalIndices;
+    }
```

Comment:
maybe add a test to ShardSearchRequestTests that makes sure that this copy ctor doesn't forget any fields? (e.g. by creating a random instance, calling this ctor and comparing them for equality?)

Reply:
++, I pushed https://github.com/elastic/elasticsearch/pull/54044/commits/a1706be437e2cecf3191592d0ea08e8017595585